packages/plugin-essentials/sources/commands/_entry.ts

```diff
@@ -16,15 +16,16 @@ export default (concierge: any) => concierge
     // berry --help
     } else if (args.length === 1 && (args[0] === `--help` || args[0] === `-h`)) {
       concierge.usage(env.argv0, {stream: stdout});
-    
+
     // berry --frozen-lockfile
     } else if (args.length === 0 || args[0].charAt(0) === `-`) {
       return await concierge.run(null, [`install`, ... args], {cwd, stdout, ... env});
-    
+
     // berry ~/projects/foo install
     } else if (args.length !== 0 && args[0].match(/[\\\/]/)) {
+      // TODO Call NodeFS.fromPortablePath
       return await concierge.run(null, args.slice(1), {cwd: posix.resolve(cwd, args[0]), stdout, ... env});
```

Comment:
What should I do here? I'm a little bit lost here 😄

Reply:
So here we detect whether the first argument looks like a path and, if it does, then we run the specified command in the specified location (instead of the current one). Since the path may be relative, we use `posix.resolve(cwd, path)` to make sure we end up with an absolute path.

Since `cwd` is already a portable path we only need to make sure that the user-provided path also is one, so in this case using `NodeFS.toPortablePath(args[0])` should be enough.

I'm not too sure whether this behavior is a good thing tbh, it's a bit confusing with regard to how the `yarn-path` directive is interpreted 🤔 But I guess this can be for a future PR.

Reply:
Ok, thanks for the explaination, I added the `toPortablePath(args[0])`